spago.lock

```diff
@@ -38,6 +39,11 @@ workspace:
       test_dependencies: []
     docs-search-common:
       path: docs-search/common
+      needed_by:
+        - spago-bin
+        - spago
+        - docs-search-index
+        - docs-search-client-halogen
```

Comment:
Both the workspace packages and the plan packages now include a `needed_by` key, so we don't even spend time computing a plan (which takes time), and instead allows for a blanket approach of "read everything in, filter for what we need, and that's our plan"

Reply:
I'm not sure I'm following this — if we list the dependencies for each package (including workspace ones), what is the advantage of needed_by?

In other words what is the difference between a) building a workspace package by looking up each of its dependencies in turn vs. b) build it by filtering by dependencies that list it in their needed_by? Aren't they equivalent?

Reply:
For example, lockfiles like `pnpm`'s list dependencies at their resolved version for every package:
https://lfx.rushstack.io/pages/concepts/pnpm_lockfile

Reply:
> if we list the dependencies for each package (including workspace ones), what is the advantage of needed_by?

The list of dependencies is just one layer of the tree, and you'd have to keep doing lookups until you reach the bottom of the tree (i.e. packages with no dependencies). Having a `needed_by` allows for storing the whole plan in there, instead of a single layer.

Reply:
I see. Would the same be achieved by listing the full resolutions for packages instead of just their explicit listed dependencies? That way a build plan for e.g. the `spago-bin` package is already stored as its list of resolutions.

Reply:
So you'd have a `resolutions` key for each workspace package listing its build plan?

Reply:
Makes sense, let me try it out